.github/workflows/iwyu.yml

```diff
@@ -0,0 +1,145 @@
+      contents: write
+      pull-requests: write
+    runs-on: ubuntu-22.04
+    timeout-minutes: 180
+    steps:
+      - name: Checkout Orbit
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: '0'
+          token: '${{ secrets.ORBITPROFILER_BOT_PAT }}'
+          path: orbit
+      - name: Setup git
+        working-directory: ./orbit
+        run: |
+          git config user.name orbitprofiler-bot
+          git config user.email orbitprofiler-bot@google.com
+      - name: Add llvm-15 repo
+        run: |
+          sudo bash -c 'echo "deb http://apt.llvm.org/jammy/ llvm-toolchain-jammy-15 main" >> /etc/apt/sources.list'
+          sudo bash -c 'echo "deb-src http://apt.llvm.org/jammy/ llvm-toolchain-jammy-15 main" >> /etc/apt/sources.list'
+          wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key|sudo apt-key add -
+      - name: Install dependencies
+        run: |
+           sudo apt-get update &&                             \
+           sudo apt-get install --yes --no-install-recommends \
+           build-essential                                    \
+           cmake                                              \
+           ccache                                             \
+           libboost-dev                                       \
+           libcapstone-dev                                    \
+           libgrpc++-dev                                      \
+           libssh2-1-dev                                      \
+           vulkan-validationlayers-dev                        \
+           libz-dev                                           \
+           llvm-dev                                           \
+           protobuf-compiler-grpc                             \
+           pkg-config                                         \
+           libvulkan-volk-dev                                 \
+           libvulkan-dev                                      \
+           libopengl-dev                                      \
+           libglx-dev                                         \
+           mesa-common-dev                                    \
+           qtbase5-dev                                        \
+           libgtest-dev                                       \
+           libgmock-dev                                       \
+           git                                                \
+           ninja-build                                        \
+           clang-format-14                                    \
+           llvm-15-dev                                        \
+           libclang-15-dev                                    \
+           clang-15                                           \
+           patchutils                                         \
+           libprotobuf-dev
+      - name: Checkout IWYU
+        uses: actions/checkout@v3
+        with:
+          repository: include-what-you-use/include-what-you-use
+          ref: clang_15
```

Comment:
Should we pin iwyu to a fixed version here? Otherwise we might have random build breaks, right? Also from a security perspective, I guess pinning might be a good idea.

Reply:
Totally agree. Thanks for the catch. Pinned it to the latest commit on that branch.